### ISSUE TITLE
big periods of 'loginwindow' recorded when macbook is closed

### ISSUE THREAD

**Issue by rtpHarry** [ActivityWatch/activitywatch#810]
- [x] I am on the [latest](https://github.com/ActivityWatch/activitywatch/releases/latest) ActivityWatch version.
 - [x] I have searched the issues of this repo and believe that this is not a duplicate.

- **OS name and version**: macOS Monterey 12.6 (21G115)
- **ActivityWatch version**: v0.12.1

## Describe the bug

Since the latest version it is picking up `loginwindow` when my laptop is closed:

<img width="1151" alt="image" src="https://user-images.githubusercontent.com/1038062/195994544-4f8e2245-a78f-4b2d-871d-2b5ab7d5b422.png">

Looks like AFK is still registering ok:

<img width="1156" alt="image" src="https://user-images.githubusercontent.com/1038062/195994635-9cfd5e18-cabc-4e3f-93e8-51878b1dcc41.png">

Don't know if this is classed as a bug, bit it feels like one, and is different behaviour to v0.11

**Comment by BelKed** [comment]
I can reproduce this "bug" on the same OS and version.

The cause of this "bug" is the usage of another windows watching strategy implemented in `0.12.1` (https://github.com/ActivityWatch/aw-watcher-window/commit/fc423c1d95e65b269e23d9f424fc7bfc38777e52, https://github.com/ActivityWatch/aw-watcher-window/commit/6d68ae35353bd43e8f0c7235a2a613dcfb272c17).

**Comment by cachedout** [comment]
I can also confirm this behavior using ActivityWatch v0.12.1 with MacOS Ventura 13.2 and it certainly feels like a bug to me. IMHO it should be quite clear that if the laptop is shut, no windows should be monitored.

@ErikBjare Since it looks like you were in the code mentioned by @BelKed , is there any chance that a fix might be on your radar? :)

**Comment by ErikBjare** [comment]
Yeah this is kind of a bug, althought doesn't really impact normal use (unless you look at data that isn't AFK filtered).

It's ultimately a side-effect of switching to event-based window watching (assuming windows stay active until we get notified otherwise) over polling (checking what the active window is every x sec). Me and @iloveitaly have discussed adding a "hybrid polling" method to the new window watching strategy to fix another issue (where window-switch events are not triggered for some reason), which could solve this issue at the same time. (can't find original discussion, but was mentioned by @modderme123 here: https://github.com/ActivityWatch/aw-watcher-window/pull/69#issuecomment-1264419382)

**Comment by modderme123** [comment]
I used https://github.com/ActivityWatch/aw-watcher-window/pull/68#discussion_r914355460 in the past, but I think we deleted that?

**Comment by cachedout** [comment]
> Yeah this is kind of a bug, althought doesn't really impact normal use (unless you look at data that isn't AFK filtered).

FIltering by AFK seems to be a fairly good workaround on the `Activity` page but this bug is still quite noticeable on the `Timeline` page. (I think this was implicit in your point, but I just wanted to call it out specifically.) Thanks for the reply and for the project. It's quite good!

**Comment by 000000ooo** [comment]
![Image](https://github.com/user-attachments/assets/68391cc2-4eda-44d2-82d2-6b1ff21a4259)
It seems to be the case that the afk monitor is also picking up on the loginwindow despite clearly being AFK. I can't imagine myself wriggling my mouse every few minutes in the mac login screen.
